intel-sgx/sgxs-tools/src/bin/sgxs-load.rs

```diff
@@ -29,20 +28,24 @@ use sgxs_loaders::enclaveapi::Sgx as SgxDevice;
 fn enclu_eenter(tcs: &mut dyn Tcs) {
     let result: u32;
     unsafe {
-        llvm_asm!("
-        lea aep(%rip),%rcx
-        jmp enclu
-aep:
-        xor %eax,%eax
-        jmp post
-enclu:
-        enclu
-post:
-"       : "={eax}"(result)
-            : "{eax}"(Enclu::EEnter), "{rbx}"(tcs.address())
-            : "rcx"
-            : "volatile"
-        )
+        std::arch::asm!("
+            xchg %rbx, {0}
+            lea 1f(%rip),%rcx
+            jmp 2f
+1:
+            xor %eax,%eax
+            jmp 3f
+2:
+            enclu
+3:
+            xchg {0}, %rbx
+",
+            // rbx is used internally by LLVM and cannot be used as an operand for inline asm (#84658)
+            in(reg) tcs.address(),
+            inout("eax") Enclu::EEnter as u32 => result,
+            lateout("rcx") _,
+            options(nostack, att_syntax)
+        );
```

Comment:
Can't this be written as:
```
            xchg %rbx, {0}
            lea aep(%rip),%rcx
            enclu
            jmp post
aep:
            xor %eax,%eax
post:
            xchg {0}, %rbx
```
it'd save a jump

Reply:
Possibly but I didn't want to make any functional changes in this PR